src/Gaufrette/Adapter/GoogleCloudClientStorage.php

```diff
@@ -0,0 +1,340 @@
+     * 
+     * @param   array   $options
+     */
+    public function setOptions($options)
+    {
+        $this->options = array_replace($this->options, $options);
+    }
+    
+    protected function computePath($key)
+    {
+        if (strlen($this->options['directory']))
+        {
+            if (strcmp(substr($this->options['directory'], -1), '/') == 0)
+            {
+                return $this->options['directory'].$key;
+            }
+            return $this->options['directory'].'/'.$key;
+        }
+        return $key;
+    }
+    
+    protected function isBucket()
+    {
+        if ($this->bucketValidated === true)
+        {
+            return true;
+        } elseif (!$this->bucket->exists()) {
+            throw new \RuntimeException(sprintf('Bucket %s does not exist.', $this->bucket->name()));
+        }
+        $this->bucketValidated = true;
+        return true;
+    }
+    
+    public function setBucket($name)
+    {
+        $this->bucketValidated = null;
+        $this->bucket = $this->storageClient->bucket($name);       
+        $this->isBucket();
+    }
+    
+    public function getBucket()
+    {
+        return $this->bucket;
+    }
+    
+    /**
+     * {@inheritdoc}
+     */
+    public function read($key)
+    {   
+        $this->isBucket();     
+        $object = $this->bucket->object($this->computePath($key));
+        $info = $object->info();
+        $this->setResources($key, $info);
+        return $object->downloadAsString();
+    }
+    
+    /**
+     * {@inheritdoc}
+     */
+    public function write($key, $content)
+    {
+        $this->isBucket();
+        
+        $options = array(
+            'resumable'     => true,
+            'name'          => $this->computePath($key),
+            'metadata'      => $this->getMetadata($key),
+        );
+
+        $this->bucket->upload(
+            $content,
+            $options
+        );
+        
+        $this->updateKeyProperties($key,
+            array(
+                'acl'       => $this->options['acl'],
+                'metadata'  => $this->getMetadata($key)
+            )
+        );
+        
+        $size = $this->getResourceByName($key, 'size');
+        
+        return $size === null ? false : $size;
+    }
+    
+    /**
+     * {@inheritdoc}
+     */
+    public function exists($key)
+    {
+        $this->isBucket();
+        $object = $this->bucket->object($this->computePath($key));
+        if ($object->exists())
+        {
+            return true;
+        }
+        return false;
+    }
+    
+    /**
+     * {@inheritdoc}
+     */
+    public function isDirectory($key)
+    {
+        if ($this->exists($key . '/'))
+        {
+            return true;
+        }
+        return false;
+    }
+    
+    /**
+     * {@inheritdoc}
+     */
+    public function listKeys($prefix = null)
+    {
+        $this->isBucket();        
+        $keys = array();        
+        if ($prefix === null)
```

Comment:
Why don't you allow null value for `computePath` so you won't have two ways of initializing `$prefix`? Btw, it's considered as a bad practice to initialize variables in if/else without default value outside: variables should be defined, at most, at the same level than they are used.